rct229/ruleset_functions/baseline_systems/is_baseline_system_7.py

```diff
@@ -0,0 +1,154 @@
+    purchased CHW), system 7b (system 7 with purchased heating), pr system 7c (system 7 with purchased heating and
+    purchased CHW). -------
+    """
+    is_baseline_system_7_str = NOT_SYS_7
+
+    # Get the hvac system
+    hvac_b = find_exactly_one_with_field_value(
+        "buildings[*].building_segments[*].heating_ventilation_air_conditioning_systems",
+        "id",
+        hvac_b_id,
+        rmi_b,
+    )
+
+    # check if the hvac system has the required sub systems for system type 7
+    has_required_sys = (
+        hvac_b.get("preheat_system") is not None
+        and hvac_b.get("cooling_system") is not None
+        and
+        # heating system shall be None or heating system type shall be None (the heating_system_type =
+        # HEATING_SYSTEM.NONE)
+        (
+            hvac_b.get("heating_system") is None
+            or find_one(
+                f"heating_system[?(@.heating_system_type={HEATING_SYSTEM.NONE})]",
+                hvac_b,
+            )
+            is not None
+        )
+    )
+
+    are_sys_data_matched = (
+        # sub functions handles missing required sys, and return False.
+        is_hvac_sys_preheating_type_fluid_loop(rmi_b, hvac_b_id)
+        and is_hvac_sys_cooling_type_fluid_loop(rmi_b, hvac_b_id)
+        and is_hvac_sys_fan_sys_vsd(rmi_b, hvac_b_id)
+        and does_each_zone_have_only_one_terminal(rmi_b, zone_id_list)
+        and are_all_terminal_heat_sources_hot_water(rmi_b, terminal_unit_id_list)
+        and are_all_terminal_cool_sources_none_or_null(rmi_b, terminal_unit_id_list)
+        and are_all_terminal_fans_null(rmi_b, terminal_unit_id_list)
+        and are_all_terminal_types_VAV(rmi_b, terminal_unit_id_list)
+    )
+
+    if has_required_sys and are_sys_data_matched:
+        # Confirm required data for Sys-7, now to decide which system type 7
+        is_hvac_sys_preheat_fluid_loop_attached_to_boiler_flag = (
+            is_hvac_sys_preheat_fluid_loop_attached_to_boiler(rmi_b, hvac_b_id)
+        )
+        are_all_terminal_heating_loops_attached_to_boiler_flag = (
+            are_all_terminal_heating_loops_attached_to_boiler(
+                rmi_b, terminal_unit_id_list
+            )
+        )
+        is_hvac_sys_fluid_loop_attached_to_chiller_flag = (
+            is_hvac_sys_fluid_loop_attached_to_chiller(rmi_b, hvac_b_id)
+        )
+        is_hvac_sys_fluid_loop_purchased_chw_flag = (
+            is_hvac_sys_fluid_loop_purchased_chw(rmi_b, hvac_b_id)
+        )
+        is_hvac_sys_preheat_fluid_loop_purchased_heating_flag = (
+            is_hvac_sys_preheat_fluid_loop_purchased_heating(rmi_b, hvac_b_id)
+        )
+        are_all_terminal_heating_loops_purchased_heating_flag = (
+            are_all_terminal_heating_loops_purchased_heating(
+                rmi_b, terminal_unit_id_list
+            )
+        )
+        if (
+            is_hvac_sys_preheat_fluid_loop_attached_to_boiler_flag
```

Comment:
Since these two flags are only used once, and because the flag names are essentially the same as the function names, I suggest calling the functions directly in the if else structure and dispense with the flags. So:
```
if is_hvac_sys_preheat_fluid_loop_attached_to_boiler(rmi_b, hvac_b_id) and 
  are_all_terminal_heating_loops_attached_to_boiler(mi_b, terminal_unit_id_list):
```
Same for is_hvac_sys_preheat_fluid_loop_purchased_heating_flag
            and are_all_terminal_heating_loops_purchased_heating_flag below.